UI/EnemyTree.gd

```diff
@@ -30,16 +30,35 @@ func _on_FilterLineEdit_text_changed(new_text):
 	_rebuild_list(new_text)
 	
 func _rebuild_list(filter_text: String = ""):
-	var normalized_filter_text := filter_text.to_upper()
+	var normalized_filter_text = filter_text.to_upper()
+	var filter_int = int(filter_text)
 	clear()
 	hide_root = true
-	var root := create_item()
+	var root = create_item()
+
 	for enemy in enemy_cache:
-		if normalized_filter_text.length() == 0 or normalized_filter_text in enemy.name.to_upper():
-			var enemy_item := create_item(root)
-			enemy_item.set_text(0, enemy.name)
-			enemy_item.set_metadata(0, enemy)
-	
+		# Defining the hex value by back converting to hex
+		var enemy_hex = int("0x%06X" % enemy.id)
```

Comment:
since this is only used to compare when the filter text is an hex number, you could do this conversion inside that elif